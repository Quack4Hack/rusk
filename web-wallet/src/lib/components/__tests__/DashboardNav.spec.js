import { afterEach, describe, expect, it } from "vitest";
import { cleanup, render } from "@testing-library/svelte";
import { DashboardNav } from "..";
import { mdiPlusBoxOutline, mdiSwapHorizontal, mdiTimerSand } from "@mdi/js";

describe("DashboardNav", () => {
  const baseProps = {
    items: [
      {
        href: "#",
        id: "item-1",
        label: "Something",
      },
      {
        href: "#",
        icons: [],
        id: "item-2",
        label: "Send",
      },
      {
        href: "#",
        icons: [{ path: mdiTimerSand }],
        id: "item-3",
        label: "Receive",
      },
      {
        href: "#",
        icons: [
          { path: mdiTimerSand },
          { path: mdiSwapHorizontal },
          { path: mdiPlusBoxOutline },
        ],
        id: "item-4",
        label: "Stake",
      },
    ],
  };
  const baseOptions = {
    props: baseProps,
    target: document.body,
  };

  afterEach(cleanup);

  it("renders the DashboardNav component", () => {
    const { container } = render(DashboardNav, baseOptions);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should not display icons if the item's icons array is undefined", () => {
    const { container } = render(DashboardNav, baseProps);

    expect(
      container.querySelector(
        ".dashboard-nav > .dashboard-nav__item:nth-child(1) > .dashboard-nav__item-icons"
      )
    ).toBeNull();

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should not display icons if the item's icon array is empty", () => {
    const { container } = render(DashboardNav, baseProps);

    expect(
      container.querySelector(
        ".dashboard-nav > .dashboard-nav__item:nth-child(2) > .dashboard-nav__item-icons"
      )
    ).toBeNull();

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should display icons if the item's icon array is present", () => {
    const { container } = render(DashboardNav, baseProps);

    expect(
      container.querySelector(
        ".dashboard-nav > .dashboard-nav__item:nth-child(3) > .dashboard-nav__item-icons"
      )
    ).toBeTruthy();

    expect(container.firstChild).toMatchSnapshot();
  });
});
