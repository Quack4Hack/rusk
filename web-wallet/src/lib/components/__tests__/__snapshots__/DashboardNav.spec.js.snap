// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DashboardNav > renders the DashboardNav component 1`] = `
<nav
  class="dashboard-nav svelte-5szabu"
>
  <ul
    class="dashboard-nav-list svelte-5szabu"
  >
    <li>
      <a
        class="dusk-anchor dusk-anchor--on-surface dashboard-nav-list__item"
        href="#"
      >
        <span
          class="dashboard-nav-item-label svelte-5szabu"
        >
          Something
        </span>
         
        
      </a>
       
    </li>
    <li>
      <a
        class="dusk-anchor dusk-anchor--on-surface dashboard-nav-list__item"
        href="#"
      >
        <span
          class="dashboard-nav-item-label svelte-5szabu"
        >
          Send
        </span>
         
        
      </a>
       
    </li>
    <li>
      <a
        class="dusk-anchor dusk-anchor--on-surface dashboard-nav-list__item"
        href="#"
      >
        <span
          class="dashboard-nav-item-label svelte-5szabu"
        >
          Receive
        </span>
         
        <span
          class="dashboard-nav-item-icons svelte-5szabu"
        >
          
          <svg
            class="dusk-icon dusk-icon--size--default"
            role="graphics-symbol"
            viewBox="0 0 24 24"
          >
            <path
              d="M6,2H18V8H18V8L14,12L18,16V16H18V22H6V16H6V16L10,12L6,8V8H6V2M16,16.5L12,12.5L8,16.5V20H16V16.5M12,11.5L16,7.5V4H8V7.5L12,11.5M10,6H14V6.75L12,8.75L10,6.75V6Z"
            />
          </svg>
          
        </span>
        
      </a>
       
    </li>
    <li>
      <a
        class="dusk-anchor dusk-anchor--on-surface dashboard-nav-list__item"
        href="#"
      >
        <span
          class="dashboard-nav-item-label svelte-5szabu"
        >
          Stake
        </span>
         
        <span
          class="dashboard-nav-item-icons svelte-5szabu"
        >
          
          <svg
            class="dusk-icon dusk-icon--size--default"
            role="graphics-symbol"
            viewBox="0 0 24 24"
          >
            <path
              d="M6,2H18V8H18V8L14,12L18,16V16H18V22H6V16H6V16L10,12L6,8V8H6V2M16,16.5L12,12.5L8,16.5V20H16V16.5M12,11.5L16,7.5V4H8V7.5L12,11.5M10,6H14V6.75L12,8.75L10,6.75V6Z"
            />
          </svg>
          
          
          <svg
            class="dusk-icon dusk-icon--size--default"
            role="graphics-symbol"
            viewBox="0 0 24 24"
          >
            <path
              d="M21,9L17,5V8H10V10H17V13M7,11L3,15L7,19V16H14V14H7V11Z"
            />
          </svg>
          
          
          <svg
            class="dusk-icon dusk-icon--size--default"
            role="graphics-symbol"
            viewBox="0 0 24 24"
          >
            <path
              d="M19,19V5H5V19H19M19,3A2,2 0 0,1 21,5V19A2,2 0 0,1 19,21H5A2,2 0 0,1 3,19V5C3,3.89 3.9,3 5,3H19M11,7H13V11H17V13H13V17H11V13H7V11H11V7Z"
            />
          </svg>
          
        </span>
        
      </a>
       
    </li>
  </ul>
</nav>
`;

exports[`DashboardNav > should display icons if the item's icon array is present 1`] = `
<nav
  class="dashboard-nav svelte-5szabu"
>
  <ul
    class="dashboard-nav-list svelte-5szabu"
  >
    <li>
      <a
        class="dusk-anchor dusk-anchor--on-surface dashboard-nav-list__item"
        href="#"
      >
        <span
          class="dashboard-nav-item-label svelte-5szabu"
        >
          Something
        </span>
         
        
      </a>
       
    </li>
    <li>
      <a
        class="dusk-anchor dusk-anchor--on-surface dashboard-nav-list__item"
        href="#"
      >
        <span
          class="dashboard-nav-item-label svelte-5szabu"
        >
          Send
        </span>
         
        
      </a>
       
    </li>
    <li>
      <a
        class="dusk-anchor dusk-anchor--on-surface dashboard-nav-list__item"
        href="#"
      >
        <span
          class="dashboard-nav-item-label svelte-5szabu"
        >
          Receive
        </span>
         
        <span
          class="dashboard-nav-item-icons svelte-5szabu"
        >
          
          <svg
            class="dusk-icon dusk-icon--size--default"
            role="graphics-symbol"
            viewBox="0 0 24 24"
          >
            <path
              d="M6,2H18V8H18V8L14,12L18,16V16H18V22H6V16H6V16L10,12L6,8V8H6V2M16,16.5L12,12.5L8,16.5V20H16V16.5M12,11.5L16,7.5V4H8V7.5L12,11.5M10,6H14V6.75L12,8.75L10,6.75V6Z"
            />
          </svg>
          
        </span>
        
      </a>
       
    </li>
    <li>
      <a
        class="dusk-anchor dusk-anchor--on-surface dashboard-nav-list__item"
        href="#"
      >
        <span
          class="dashboard-nav-item-label svelte-5szabu"
        >
          Stake
        </span>
         
        <span
          class="dashboard-nav-item-icons svelte-5szabu"
        >
          
          <svg
            class="dusk-icon dusk-icon--size--default"
            role="graphics-symbol"
            viewBox="0 0 24 24"
          >
            <path
              d="M6,2H18V8H18V8L14,12L18,16V16H18V22H6V16H6V16L10,12L6,8V8H6V2M16,16.5L12,12.5L8,16.5V20H16V16.5M12,11.5L16,7.5V4H8V7.5L12,11.5M10,6H14V6.75L12,8.75L10,6.75V6Z"
            />
          </svg>
          
          
          <svg
            class="dusk-icon dusk-icon--size--default"
            role="graphics-symbol"
            viewBox="0 0 24 24"
          >
            <path
              d="M21,9L17,5V8H10V10H17V13M7,11L3,15L7,19V16H14V14H7V11Z"
            />
          </svg>
          
          
          <svg
            class="dusk-icon dusk-icon--size--default"
            role="graphics-symbol"
            viewBox="0 0 24 24"
          >
            <path
              d="M19,19V5H5V19H19M19,3A2,2 0 0,1 21,5V19A2,2 0 0,1 19,21H5A2,2 0 0,1 3,19V5C3,3.89 3.9,3 5,3H19M11,7H13V11H17V13H13V17H11V13H7V11H11V7Z"
            />
          </svg>
          
        </span>
        
      </a>
       
    </li>
  </ul>
</nav>
`;

exports[`DashboardNav > should not display icons if the item's icon array is empty 1`] = `
<nav
  class="dashboard-nav svelte-5szabu"
>
  <ul
    class="dashboard-nav-list svelte-5szabu"
  >
    <li>
      <a
        class="dusk-anchor dusk-anchor--on-surface dashboard-nav-list__item"
        href="#"
      >
        <span
          class="dashboard-nav-item-label svelte-5szabu"
        >
          Something
        </span>
         
        
      </a>
       
    </li>
    <li>
      <a
        class="dusk-anchor dusk-anchor--on-surface dashboard-nav-list__item"
        href="#"
      >
        <span
          class="dashboard-nav-item-label svelte-5szabu"
        >
          Send
        </span>
         
        
      </a>
       
    </li>
    <li>
      <a
        class="dusk-anchor dusk-anchor--on-surface dashboard-nav-list__item"
        href="#"
      >
        <span
          class="dashboard-nav-item-label svelte-5szabu"
        >
          Receive
        </span>
         
        <span
          class="dashboard-nav-item-icons svelte-5szabu"
        >
          
          <svg
            class="dusk-icon dusk-icon--size--default"
            role="graphics-symbol"
            viewBox="0 0 24 24"
          >
            <path
              d="M6,2H18V8H18V8L14,12L18,16V16H18V22H6V16H6V16L10,12L6,8V8H6V2M16,16.5L12,12.5L8,16.5V20H16V16.5M12,11.5L16,7.5V4H8V7.5L12,11.5M10,6H14V6.75L12,8.75L10,6.75V6Z"
            />
          </svg>
          
        </span>
        
      </a>
       
    </li>
    <li>
      <a
        class="dusk-anchor dusk-anchor--on-surface dashboard-nav-list__item"
        href="#"
      >
        <span
          class="dashboard-nav-item-label svelte-5szabu"
        >
          Stake
        </span>
         
        <span
          class="dashboard-nav-item-icons svelte-5szabu"
        >
          
          <svg
            class="dusk-icon dusk-icon--size--default"
            role="graphics-symbol"
            viewBox="0 0 24 24"
          >
            <path
              d="M6,2H18V8H18V8L14,12L18,16V16H18V22H6V16H6V16L10,12L6,8V8H6V2M16,16.5L12,12.5L8,16.5V20H16V16.5M12,11.5L16,7.5V4H8V7.5L12,11.5M10,6H14V6.75L12,8.75L10,6.75V6Z"
            />
          </svg>
          
          
          <svg
            class="dusk-icon dusk-icon--size--default"
            role="graphics-symbol"
            viewBox="0 0 24 24"
          >
            <path
              d="M21,9L17,5V8H10V10H17V13M7,11L3,15L7,19V16H14V14H7V11Z"
            />
          </svg>
          
          
          <svg
            class="dusk-icon dusk-icon--size--default"
            role="graphics-symbol"
            viewBox="0 0 24 24"
          >
            <path
              d="M19,19V5H5V19H19M19,3A2,2 0 0,1 21,5V19A2,2 0 0,1 19,21H5A2,2 0 0,1 3,19V5C3,3.89 3.9,3 5,3H19M11,7H13V11H17V13H13V17H11V13H7V11H11V7Z"
            />
          </svg>
          
        </span>
        
      </a>
       
    </li>
  </ul>
</nav>
`;

exports[`DashboardNav > should not display icons if the item's icons array is undefined 1`] = `
<nav
  class="dashboard-nav svelte-5szabu"
>
  <ul
    class="dashboard-nav-list svelte-5szabu"
  >
    <li>
      <a
        class="dusk-anchor dusk-anchor--on-surface dashboard-nav-list__item"
        href="#"
      >
        <span
          class="dashboard-nav-item-label svelte-5szabu"
        >
          Something
        </span>
         
        
      </a>
       
    </li>
    <li>
      <a
        class="dusk-anchor dusk-anchor--on-surface dashboard-nav-list__item"
        href="#"
      >
        <span
          class="dashboard-nav-item-label svelte-5szabu"
        >
          Send
        </span>
         
        
      </a>
       
    </li>
    <li>
      <a
        class="dusk-anchor dusk-anchor--on-surface dashboard-nav-list__item"
        href="#"
      >
        <span
          class="dashboard-nav-item-label svelte-5szabu"
        >
          Receive
        </span>
         
        <span
          class="dashboard-nav-item-icons svelte-5szabu"
        >
          
          <svg
            class="dusk-icon dusk-icon--size--default"
            role="graphics-symbol"
            viewBox="0 0 24 24"
          >
            <path
              d="M6,2H18V8H18V8L14,12L18,16V16H18V22H6V16H6V16L10,12L6,8V8H6V2M16,16.5L12,12.5L8,16.5V20H16V16.5M12,11.5L16,7.5V4H8V7.5L12,11.5M10,6H14V6.75L12,8.75L10,6.75V6Z"
            />
          </svg>
          
        </span>
        
      </a>
       
    </li>
    <li>
      <a
        class="dusk-anchor dusk-anchor--on-surface dashboard-nav-list__item"
        href="#"
      >
        <span
          class="dashboard-nav-item-label svelte-5szabu"
        >
          Stake
        </span>
         
        <span
          class="dashboard-nav-item-icons svelte-5szabu"
        >
          
          <svg
            class="dusk-icon dusk-icon--size--default"
            role="graphics-symbol"
            viewBox="0 0 24 24"
          >
            <path
              d="M6,2H18V8H18V8L14,12L18,16V16H18V22H6V16H6V16L10,12L6,8V8H6V2M16,16.5L12,12.5L8,16.5V20H16V16.5M12,11.5L16,7.5V4H8V7.5L12,11.5M10,6H14V6.75L12,8.75L10,6.75V6Z"
            />
          </svg>
          
          
          <svg
            class="dusk-icon dusk-icon--size--default"
            role="graphics-symbol"
            viewBox="0 0 24 24"
          >
            <path
              d="M21,9L17,5V8H10V10H17V13M7,11L3,15L7,19V16H14V14H7V11Z"
            />
          </svg>
          
          
          <svg
            class="dusk-icon dusk-icon--size--default"
            role="graphics-symbol"
            viewBox="0 0 24 24"
          >
            <path
              d="M19,19V5H5V19H19M19,3A2,2 0 0,1 21,5V19A2,2 0 0,1 19,21H5A2,2 0 0,1 3,19V5C3,3.89 3.9,3 5,3H19M11,7H13V11H17V13H13V17H11V13H7V11H11V7Z"
            />
          </svg>
          
        </span>
        
      </a>
       
    </li>
  </ul>
</nav>
`;
